pkg/kapp/app/apps.go

```diff
@@ -89,7 +90,7 @@ func (a Apps) list(additionalLabels map[string]string, nsName string) ([]App, er
 		name := app.Name
 		isMigrated := false
 
-		if _, ok := app.Annotations[KappIsConfigmapMigratedAnnotationKey]; ok {
+		if _, ok := app.Annotations[KappIsConfigmapMigratedAnnotationKey]; ok && strings.ToLower(os.Getenv("KAPP_FQ_CONFIGMAP_NAMES")) == "true" {
```

Comment:
can we not use isMigrationEnabled() here?

Reply:
That sounds better!

Reply:
it's probably time to make a constructor for RecorderApp: https://github.com/vmware-tanzu/carvel-kapp/blob/develop/pkg/kapp/app/apps.go#L92-L99. seems like a log of app guts are in Apps

Reply:
> seems like a log of app guts are in Apps

Yep, I have been thinking the same for a while.

> it's probably time to make a constructor for RecorderApp

Makes sense.

Reply:
Apps looks fine now, but I feel that the Recorded app constructor is a bit complex now (with the configmap and the usage of private methods)